packages/trader/src/App/Containers/Modals/__tests__/trade-modals.spec.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { mockStore } from '@deriv/stores';
+import TraderProviders from '../../../../trader-providers';
+import TradeModals from '../trade-modals';
+
+jest.mock('App/Components/Elements/Modals/UnsupportedContractModal', () =>
+    jest.fn(props => (
+        <div>
+            <div>Unsupported contract modal</div>
+            <button onClick={props.onConfirm}>onConfirm unsupported</button>
+            <button onClick={props.onClose}>onClose unsupported</button>
+        </div>
+    ))
+);
+jest.mock('App/Components/Elements/Modals/MarketUnavailableModal', () =>
+    jest.fn(props => (
+        <div>
+            <div>Market unavailable modal</div>
+            <button onClick={props.onConfirm}>onConfirm market</button>
+            <button onClick={props.onCancel}>onCancel market</button>
+        </div>
+    ))
+);
+jest.mock('App/Components/Elements/Modals/ServicesErrorModal', () =>
+    jest.fn(props => (
+        <div>
+            <div>Services error modal</div>
+            <button onClick={props.onConfirm}>onConfirm services</button>
+        </div>
+    ))
+);
+
+window.open = jest.fn();
+
+describe('TradeModals', () => {
+    let mock_root_store = mockStore({});
+
+    beforeEach(() => {
+        mock_root_store = mockStore({
+            modules: {
+                trade: {
+                    resetPreviousSymbol: jest.fn(),
+                    clearPurchaseInfo: jest.fn(),
+                    requestProposal: jest.fn(),
+                },
+            },
+        });
+    });
```

Comment:
@kate-deriv Please use the `mockStore` inside each test case 🙏🏻

Reply:
@kate-deriv In order to remove duplication while still initializing mockStore in every test case, please use an approach used in function mockOnRamp() in packages/cashier/src/pages/on-ramp/__tests__/on-ramp.spec.tsx in order to incapsulate TraderProviders like it's done there for CashierProviders.